### PR TITLE
JDK-8234128 jarsigner will not show not-signed-by-alias warning if an intermediate cert is in this keystore

### PR DESCRIPTION
**Pull request by mcpowers**
https://bugs.openjdk.java.net/browse/JDK-8234128

My initial attempt had too many bad side-effects. This latest fix only tries to suppress the warning message.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8234128](https://bugs.openjdk.java.net/browse/JDK-8234128): jarsigner will not show not-signed-by-alias warning if an intermediate cert is in this keystore

### Reviewers
 * [Jamil Nimeh](https://openjdk.java.net/census#jnimeh) (@jnimeh - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7751/head:pull/7751` \
`$ git checkout pull/7751`

Update a local copy of the PR: \
`$ git checkout pull/7751` \
`$ git pull https://git.openjdk.java.net/jdk pull/7751/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7751`

View PR using the GUI difftool: \
`$ git pr show -t 7751`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7751.diff">https://git.openjdk.java.net/jdk/pull/7751.diff</a>

</details>

### CODE REVIEW

File: src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java [hunk 1]
```diff
@@ -222,6 +223,7 @@ public static void main(String args[]) throws Exception {
     private boolean badExtendedKeyUsage = false;
     private boolean badNetscapeCertType = false;
     private boolean signerSelfSigned = false;
+    private boolean not_all_entries_signed_by_alias_in_keystore = false;
```
[the remark this answers is not in the document]
Comment (mcpowers) [reply src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java:226]: Good Point. This is Java. Fixed.

**Comment by mcpowers** [comment]
I recall that Max said the intermediate was not necessary. Maybe he can chime in.

**Comment by jnimeh** [comment]
Yeah, the loop seems like it would behave the same way with or without the intermediate CA cert for the purposes of the warning message.  I'm good with it.

**Review by jnimeh — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**